ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7759,6 +7759,18 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
               .getMsg(destinationPath.toUri().toString()));
         }
       }
+      // handle direct insert CTAS case
+      // for direct insert CTAS, the table creation DDL is not added to the task plan in TaskCompiler,
+      // therefore we need to add the InsertHook here manually so that HiveMetaHook#commitInsertTable is called
+      if (qb.isCTAS() && tableDesc != null && tableDesc.getStorageHandler() != null) {
+        try {
+          if (HiveUtils.getStorageHandler(conf, tableDesc.getStorageHandler()).directInsertCTAS()) {
+            createPreInsertDesc(destinationTable, false);
+          }
+        } catch (HiveException e) {
```

Comment:
When do we get `HiveException`? Why swallow it?

Reply:
See above. I was thinking of swallowing it because I just wanted to proceed with the operation even if the storage handler class is not loadable. Do you think we should throw an exception and kill the query in this case?

Reply:
I think that would make sense, as we are not able to perform the CTAS without the classes on classpath anyway.

Reply:
Right. Now that I think about it, the main reason I swallowed the exception is that this is a general hive codepath, so didn't want to screw up any normal hive ctas queries. But I think we can assume that even for normal hive ctas, if the table has a storage handler defined, it should be loadable.